tests/nodes/tests_Literal.cpp

```diff
@@ -1163,3 +1163,26 @@ TEST_CASE("Literal::fetch_or_serialize_lexical_form") {
     }
 }
 
+template<datatypes::LiteralDatatype L>
+void check_leading_zeros_parse() {
+    SUBCASE(L::identifier.c_str()) {
+        auto lit = Literal::make_typed<L>("09");
+        CHECK_FALSE(lit.null());
+        CHECK_EQ(lit.template value<L>(), 9);
+
+        lit = Literal::make_typed<L>("000");
+        CHECK_FALSE(lit.null());
+        CHECK_EQ(lit.template value<L>(), 0);
+    }
+}
+
+TEST_CASE("leading zeros parsing") {
```

Comment:
this might be a perfect case for a `TEST_CASE_TEMPLATE` (example see: https://github.com/rdf4cpp/rdf4cpp/blob/086033bdbf68e7fd49a966628796c13c04e755e9/tests/nodes/tests_Node.cpp#L304 )